src/rust-crypto/rust-crypto.ts

```diff
@@ -880,14 +874,9 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
             return;
         }
 
-        const backupKeyFromStorage = await this.secretStorage.get("m.megolm_backup.v1");
         const backupKeyBase64 = backupKeys.decryptionKey.toBase64();
 
-        // The backup version that the key corresponds to isn't saved in 4S so if it's different, we must assume
-        // it's stale and overwrite.
-        if (backupKeyFromStorage !== backupKeyBase64) {
-            await this.secretStorage.store("m.megolm_backup.v1", backupKeyBase64);
-        }
+        await this.secretStorage.store("m.megolm_backup.v1", backupKeyBase64);
```

Comment:
The documentation on this method says "... and we have secret storage active ..." - is that correct?

Reply:
Mmm, doesn't look like there's a way to determine if we have 4s set up, afaics, so I guess I'll just remove that bit.

Reply:
There is `this.secretStorageHasAESKey()` which is what `resetKeyBackup` does, but given this is only called from `bootstrapSecretStorage`, I don't think there's any need to call it.